### ISSUE TITLE
[terraform] Support lock-timeout option

### ISSUE THREAD

**Issue by steinybot** [ovotech/circleci-orbs#113]
It would be nice to have support for the `lock-timeout` option.

I have two workflows `continuous_integration` (which does a `plan`) and `continuous_deployment` (does an `apply`). The CI workflow runs on all branches. The CD workflow only runs on `master` and it is also triggered via the API from other builds which trigger a redeploy. When the build runs on `master` it runs both of these workflows concurrently and one of them will always fail as it cannot get lock because the other one has it. I need to be able to set the lock timeout so that they will retry.

**Comment by bamarch** [comment]
I can see why this would be very useful, especially if your CI/CD solution does not give you the ability to run jobs exclusively, or specify concurrency etc.

All the terraform commands you're using (implicit: init, explicit: plan,apply) support this parameter:

    -lock-timeout=0s    Duration to retry a state lock.

So adding this as a parameter and passing it through sounds good!

I'll probably take a crack at this - unless you were blocked by this and published a dev version already that implements the feature? PRs happily accepted always! :D
